spring-cloud-starter-single-step-batch-job/src/test/java/org/springframework/cloud/task/batch/autoconfigure/JdbcItemWriterAutoConfigurationTests.java

```diff
@@ -0,0 +1,261 @@
+import org.springframework.batch.item.database.support.ColumnMapItemPreparedStatementSetter;
+import org.springframework.batch.item.support.ListItemReader;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.autoconfigure.batch.BatchAutoConfiguration;
+import org.springframework.boot.autoconfigure.context.PropertyPlaceholderAutoConfiguration;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.cloud.task.batch.autoconfigure.jdbc.JdbcItemWriterAutoConfiguration;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.util.SocketUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JdbcItemWriterAutoConfigurationTests {
+
+	private final static String DATASOURCE_URL;
+
+	private final static String DATASOURCE_USER_NAME = "SA";
+
+	private final static String DATASOURCE_USER_PASSWORD = "''";
+
+	private final static String DATASOURCE_DRIVER_CLASS_NAME = "org.h2.Driver";
+
+	private static int randomPort;
+
+	static {
+		randomPort = SocketUtils.findAvailableTcpPort();
+		DATASOURCE_URL = "jdbc:h2:tcp://localhost:" + randomPort
+				+ "/mem:dataflow;DB_CLOSE_DELAY=-1;" + "DB_CLOSE_ON_EXIT=FALSE";
+	}
+
+	@Test
+	public void baseTest() {
+		ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner()
+				.withUserConfiguration(
+						JdbcItemWriterAutoConfigurationTests.DelimitedJobConfiguration.class)
+				.withConfiguration(
+						AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+								BatchAutoConfiguration.class,
+								SingleStepJobAutoConfiguration.class,
+								JdbcItemWriterAutoConfiguration.class,
+								TaskLauncherConfiguration.class));
+		applicationContextRunner = updatePropertiesForTest(applicationContextRunner);
+
+		runTest(applicationContextRunner);
+	}
+
+	@Test
+	public void customSqlParameterSourceTest() {
+		ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner()
+				.withUserConfiguration(
+						JdbcItemWriterAutoConfigurationTests.DelimitedJobConfiguration.class)
+				.withConfiguration(
+						AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+								BatchAutoConfiguration.class,
+								SingleStepJobAutoConfiguration.class,
+								JdbcItemWriterAutoConfiguration.class,
+								TaskLauncherConfiguration.class,
+								CustomSqlParameterSourceProviderConfiguration.class));
```

Comment:
`TaskLauncherConfiguration` and `CustomSqlParameterSourceProviderConfiguration` are not autoconfigurations and should be with the user configurations.

Reply:
moved